### ISSUE TITLE
Compiling 2021.8.3 using Go 1.17 fails

### ISSUE THREAD

**Issue by jauderho** [cloudflare/cloudflared#442]
This was working with no errors for 2021.8.2.

The error message is https://github.com/jauderho/dockerfiles/runs/3419574175?check_suite_focus=true#step:8:2063

Seems to be a known issue as caddyserver is reporting similar issues. See https://github.com/caddyserver/caddy/issues/4118

For now, I've reverted to building with Go 1.16.

**Comment by francislavoie** [comment]
This only requires updating quic-go to the latest version.

https://github.com/cloudflare/cloudflared/blob/4ff215ee8c133472bf95df23bd28891de84b8c10/go.mod#L32

**Comment by jauderho** [comment]
Looks like @sudarshan-reddy downgraded quic-go to 0.20.0 a few days ago in https://github.com/cloudflare/cloudflared/commit/1082ac1c36867d7e7b8e4545a6c8673827474c5f

quic-go did fix the panic in 0.21.2 which is newer than what the commit downgraded from (0.21.1)

**Comment by nmldiegues** [comment]
> Looks like @sudarshan-reddy downgraded quic-go to 0.20.0 a few days ago in [1082ac1](https://github.com/cloudflare/cloudflared/commit/1082ac1c36867d7e7b8e4545a6c8673827474c5f)

Yes, we've had to downgrade to avoid pulling https://github.com/golang/go/issues/43989 back in.
We'll check how we can reconcile both problems.

In the meantime, out of curiosity, what's driving you to compile the repo on every release?

**Comment by jauderho** [comment]
@nmldiegues Mostly as a way of learning to build minimal multi-platform images for things that I use as well as GitHub Actions. After a bunch of trials, I think I figured out a fairly clean way to build images (in particular, go based apps) to the point where I can containerize a Go binary in about 15-20 mins. 

Feel free to take a look at https://github.com/jauderho/dockerfiles or https://github.com/cloudflare/cloudflared/issues/422#issuecomment-905217253 (for details around specifically around how I build cloudflared)
